### ISSUE TITLE
Warnings on HEAD

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#2856]
```
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:113:14: warning: variable 'x' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    else if( (fabs(y) > CENTRAL_RANGE) && (fabs(y) < 1.0) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:121:9: note: uninitialized use occurs here
    x = x - (erf(x) - y)/( (2.0/sqrt(M_PI))*exp(-x*x));
        ^
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:113:10: note: remove the 'if' if its condition is always true
    else if( (fabs(y) > CENTRAL_RANGE) && (fabs(y) < 1.0) )
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:113:14: warning: variable 'x' is used uninitialized whenever '&&' condition is false [-Wsometimes-uninitialized]
    else if( (fabs(y) > CENTRAL_RANGE) && (fabs(y) < 1.0) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:121:9: note: uninitialized use occurs here
    x = x - (erf(x) - y)/( (2.0/sqrt(M_PI))*exp(-x*x));
        ^
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:113:14: note: remove the '&&' if its condition is always true
    else if( (fabs(y) > CENTRAL_RANGE) && (fabs(y) < 1.0) )
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/users/ezyang/pytorch/torch/lib/tmp_install/include/TH/THMath.h:98:13: note: initialize the variable 'x' to silence this warning
    double x,z,num,dem; /*working variables */
            ^
             = 0.0
```

**Comment by ezyang** [comment]
@IraKorshunova, you introduced `TH_ErfInv` in 2b9765ad02b666d623566935b385fc3d058ad33d; could you please take a look?

**Comment by IraKorshunova** [comment]
Thank you for pointing that out! It was a copypasted code. Let me replace  `else if` with `else` here https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/THMath.h#L113 and that should work

**Comment by ezyang** [comment]
I don't really understand how the erf implementation works, so I am not sure if putting in an else is sound. It certainly doesn't look semantics preserving!

**Comment by fmassa** [comment]
@ezyang that `else if` was unnecessary, because the previous conditionals already took care of the specific case.
The `fabs(y) < 1` is in [here](https://github.com/pytorch/pytorch/commit/2b9765ad02b666d623566935b385fc3d058ad33d#diff-6a89b015a65a10ba52ba750c92183b37R113) is already taken care in [here](https://github.com/pytorch/pytorch/commit/2b9765ad02b666d623566935b385fc3d058ad33d#diff-6a89b015a65a10ba52ba750c92183b37R104) and [here](https://github.com/pytorch/pytorch/commit/2b9765ad02b666d623566935b385fc3d058ad33d#diff-6a89b015a65a10ba52ba750c92183b37R105), and it only remains `(fabs(y) > CENTRAL_RANGE)` which is the `else` from the [first if](https://github.com/pytorch/pytorch/commit/2b9765ad02b666d623566935b385fc3d058ad33d#diff-6a89b015a65a10ba52ba750c92183b37R106)

**Comment by soumith** [comment]
fixed via #2861
